File: app/src/main/java/com/karthik/myweather/data/WeatherDatabase.java
package com.karthik.myweather.data;

import androidx.room.Database;
import androidx.room.RoomDatabase;

import com.karthik.myweather.data.dao.CityWeatherDao;
import com.karthik.myweather.data.dao.LocationEntityDao;
import com.karthik.myweather.data.entities.City;
import com.karthik.myweather.data.entities.Weather;
import com.karthik.myweather.data.entities.LocationEntity;

@Database(
        entities = {LocationEntity.class, City.class, Weather.class},
        version = 1)
public abstract class WeatherDatabase extends RoomDatabase {

    public abstract LocationEntityDao locationEntityDao();

    public abstract CityWeatherDao cityWeatherDao();
}

File: app/src/main/java/com/karthik/myweather/data/WeatherDatabase.kt
package com.karthik.myweather.data

import androidx.room.Database
import androidx.room.RoomDatabase
import com.karthik.myweather.data.dao.CityWeatherDao
import com.karthik.myweather.data.dao.LocationEntityDao
import com.karthik.myweather.data.entities.City
import com.karthik.myweather.data.entities.LocationEntity
import com.karthik.myweather.data.entities.Weather

@Database(entities = [LocationEntity::class, City::class, Weather::class], version = 1, exportSchema = false)
abstract class WeatherDatabase : RoomDatabase() {
    abstract fun locationEntityDao(): LocationEntityDao?
    abstract fun cityWeatherDao(): CityWeatherDao?
}